### PR TITLE
Bugs 2.0

### LINKED ISSUE #2958: [Bug] stocks/fa/income --source AlphaVantage no data
For aapl

```

2022 Oct 20, 19:52 (🦋) /stocks/fa/ $ income --source AlphaVantage
No data found from Alpha Vantage
```

### LINKED ISSUE #2975: [ENHANCEMENT] etf/search should add in -n if not supplied
Ie
```
search Invesco
```

### LINKED ISSUE #2976: [Bug] etf/overview "SPY" not found in ETFS?
I bet $1 its a case matching thing

etf
load spy
overview

### LINKED ISSUE #2977: [Bug] etf autocomplete?
etf
load spy
holdings ---- this autocomplete shows more than the -l/-h/--export parameters
Screen Shot 2022-10-21 at 12 48 27 PM

### LINKED ISSUE #2981: [Bug] forex load EURUSD --source AlphaVantage "No Data Loaded"
Weird.  The intraday endpoint is premium, but the daily isn't, so if we have it here, we should support it :P

```
2022 Oct 21, 16:06 (🦋) /forex/ $ load EURUSD --source AlphaVantage
Interval not supported by AlphaAdvantage. Need to be one of the following options dict_keys(['1min', '5min', '15min', '30min', '60min'])

No historical data loaded.

Make sure you have appropriate access for the 'AlphaVantage' data source and that 'AlphaVantage' supports the requested range.

EUR-USD loaded.
```

### LINKED ISSUE #2983: [Bug] forex load EURUSD -i 2min
Related: the autocomplete does not work well

```
2022 Oct 21, 16:15 (🦋) /forex/ $ load EURUSD -i 2min

1 Failed download:
- EURUSD=X: 2m data not available for startTime=1634788800 and endTime=1666383358. The requested range must be within the last 60 days.

No historical data loaded.

Make sure you have appropriate access for the 'YahooFinance' data source and that 'YahooFinance' supports the requested range.

EUR-USD loaded.
```

### PR DESCRIPTION
**Pull request by jmaslek**
Fixes #2975 -- turns out we kinda had it lol
Fixes #2976 
Fixes #2977 
Addresses/fixes #2958 :
```
2022 Oct 22, 22:51 (🦋) /stocks/fa/ $ income --source AlphaVantage
This is a premium endpoint for AlphaVantage. Please use a premium key

2022 Oct 22, 22:52 (🦋) /stocks/fa/ $ balance --source AlphaVantage
This is a premium endpoint for AlphaVantage. Please use a premium key

2022 Oct 22, 22:52 (🦋) /stocks/fa/ $ cash --source AlphaVantage
This is a premium endpoint for AlphaVantage. Please use a premium key
```

Fixes #2981 
Fixes #2983

### CODE REVIEW

File: openbb_terminal/stocks/fundamental_analysis/av_model.py [hunk 1]
```diff
@@ -19,6 +19,19 @@
 logger = logging.getLogger(__name__)
 
 
+def check_premium_key(json_response: Dict) -> bool:
+    """Checks if the response is the premium endpoint"""
+    if json_response == {
+        "Information": "Thank you for using Alpha Vantage! This is a premium endpoint. You may subscribe to "
+        "any of the premium plans at https://www.alphavantage.co/premium/ to instantly unlock all premium endpoints"
+    }:
+        console.print(
```
Comment (Chavithra) [inline openbb_terminal/stocks/fundamental_analysis/av_model.py:28]: There is probably a better way than than printing message in the model.

File: openbb_terminal/etf/etf_controller.py [hunk 2]
```diff
@@ -116,21 +116,8 @@ def __init__(self, queue: List[str] = None):
                 "--limit": one_to_fifty,
                 "-l": "--limit",
             }
-            choices["holdings"] = {
```
Comment (Chavithra) [inline openbb_terminal/etf/etf_controller.py:119]: Tried `etf/holdings` and now it show no autocompletion at all.

**Comment by Chavithra** [comment]
These commands seems ok:
- `etf/search Invesco`
- `etf/load SPY/overview`
- `stocks/fa/income --source AlphaVantage`
- `forex/load/EURUSD --source AlphaVantage`
- `forex/load EURUSD -i 2min`

Although should we have autocompletion for `load -i` ?

This one has autocompletion issue:
- `etf/load SPY/holdings`

File: openbb_terminal/etf/etf_controller.py (hunk 2 above)
Comment (jmaslek) [reply openbb_terminal/etf/etf_controller.py:119]: There aren't any autocomplete options for holdings

File: openbb_terminal/stocks/fundamental_analysis/av_model.py (hunk 1 above)
Comment (jmaslek) [reply openbb_terminal/stocks/fundamental_analysis/av_model.py:28]: There probably is but this is the way we have been doing them throughout.  @colin99d wants a consistent way

**Comment by jmaslek** [comment]
> Although shouldn't we have autocompletion for `load -i` ?

I think there currently is not any because different sources have different possible intervals.  Can add in a new issue as an enhancement, as the scope here was to fix the bug that occurred when doing the 2min flag.

> This one has autocompletion issue:
> 
> * `etf/load SPY/holdings`

There are no autocomplete options

**Review by Chavithra — APPROVED** [review]
Looks ok.

**Pull request merged**